From 2f13e3de5455f5b4597f7e7615a9e0efe53dd9ee Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Tue, 17 Jan 2017 16:37:27 +0100
Subject: [PATCH 1560/2241] arm64: dts: marvell: configure io-expander's gpio3
 as the u3d's vbus-gpio in a8k

In the Armada-8040-db board, the GPIO3 of io-expander is used as a
vbus-gpio, reflect it in device-tree description.

Change-Id: I3882757a14e47e9bda7242b26c2ff1eaa014b174
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/38543
Reviewed-by: Omri Itach <omrii@marvell.com>
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Hanna Hawa <hannah@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index ad1e2bd..effb695 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -149,6 +149,17 @@
 
 			};
 
+			u3d@520000 {
+				/* There is no need to set pinctlr, since the
+				 * gpio is from io-expander
+				 */
+
+				/* To enable IO expander interrupts, SD/MMC on CP
+				 * Must be disabled in U-Boot & Linux
+				 * vbus-gpio = <&expander0 3 GPIO_ACTIVE_HIGH>;
+				 */
+			};
+
 			/* CON9 */
 			usb3@500000 {
 				usb-phy = <&cpm_usb3h0_phy>;
-- 
2.7.4

